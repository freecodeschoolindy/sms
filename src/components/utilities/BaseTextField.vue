<template>
  <fieldset class="half">
    <label :for="id">{{ label }}</label>
    <input
      :id="id"
      v-bind="$attrs"
      :name="id"
      :type="type || 'text'"
      :value="value"
      @input="handleInput"
      v-on="getListeners"
    />
  </fieldset>
</template>

<script>
export default {
  name: 'BaseTextField',
  inheritAttrs: false,
  props: {
    label: {
      type: String,
      required: true
    },
    id: {
      type: String,
      required: true
    },
    type: {
      type: String,
      default: undefined
    },
    value: {
      type: String,
      default: undefined
    }
  },
  methods: {
    handleInput: function (event) {
      this.$emit('input', event.target.value);
    }
  },
  computed: {
    getListeners: function () {
      // eslint-disable-next-line no-unused-vars
      const { input, ...others } = this.$listeners;
      return { ...others };
    }
  }
};
</script>
