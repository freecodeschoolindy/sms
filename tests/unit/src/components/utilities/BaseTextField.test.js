import BaseTextField from '@/components/utilities/BaseTextField.vue';

import testHelpers from '@@/testHelpers.js';

describe('BaseTextField.vue', () => {
  let localVue;
  let store;
  let setup;

  let permissions = ['student'];

  beforeEach(() => {
    setup = testHelpers.setupTestVariables(permissions);
    localVue = setup.localVue;
    store = setup.store;
  });

  test('Renders correctly', async () => {
    const wrapper = await testHelpers.shallowMountWrapper(BaseTextField, {
      localVue,
      store,
      propsData: {
        id: 'id',
        label: 'label',
        value: 'value'
      }
    });

    expect(wrapper)
      .toMatchSnapshot();
  });
});
