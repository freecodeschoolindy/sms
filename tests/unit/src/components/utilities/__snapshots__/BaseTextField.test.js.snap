// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTextField.vue Renders correctly 1`] = `
<fieldset class="half">
  <label for="id">label</label>
  <input id="id" name="id" type="text">
</fieldset>
`;
